### PR TITLE
Fix ONNX `RuntimeError` for OBB models exported with NMS

### PR DESCRIPTION
**Pull request by Y-T-G**
```
yolo export model=yolo11n-obb.pt nms
yolo predict model=yolo11n-obb.onnx device=0
```

Fix for GPU CI error: https://github.com/ultralytics/ultralytics/actions/runs/17708412646/job/50323622561#step:6:3713

I just came across my own report about this on a PyTorch issue which I had forgotten about:
https://github.com/pytorch/pytorch/issues/66200#issuecomment-2849090692

This error only occurs when you run the ONNX model on CUDAExecutionProvider. The workaround is not to filter out all the boxes so that there are some remaining for NMS.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Fixes ONNX export and inference for OBB models by safely handling NMS edge cases, and re-enables corresponding tests ✅

### 📊 Key Changes
- Tests: Always pass `nms=nms` in ONNX export tests, including for OBB tasks (previously disabled) 🧪
- Exporter: Extends the safety check for empty NMS masks to ONNX when using OBB, not just TensorFlow/TFLite:
  - Applies masking logic to avoid errors when no detections pass the confidence threshold 🛡️

### 🎯 Purpose & Impact
- Improves robustness of ONNX exports with OBB + NMS, preventing crashes in edge cases (e.g., no detections) 🧷
- Restores CI coverage for OBB NMS in ONNX, catching regressions earlier ✅
- No API changes; users exporting OBB models to ONNX can now rely on stable, built-in NMS behavior 🚀

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response — an Ultralytics engineer will review and assist soon. In the meantime, please review the checklist below to help us integrate your changes smoothly:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

Since this appears to address a RuntimeError for OBB ONNX exports on CUDAExecutionProvider, please also:
- 🧪 Provide a Minimum Reproducible Example (MRE) if not already included, covering the zero-detections OBB ONNX case. Include:
  - Environment details (OS, Python, CUDA/cuDNN, ONNX Runtime provider)
  - Model and exact export/inference commands
  - A minimal input (or synthetic input) that reproduces the issue
- 🔍 Add or update a test that validates the empty-mask safeguard for OBB during ONNX export and inference, ensuring no reshape or gather errors when there are no detections

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Comment by onuralpszr** [comment]
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)<sub>

GitHub Actions below triggered via workflow dispatch on this PR branch `obb-nms-gpu` at 2025-09-15 06:28:34 UTC with `@ultralytics/run-ci` command:

* ✅ [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/runs/17724110196): `ci.yml` run 11989
* ✅ [Publish Docker Images](https://github.com/ultralytics/ultralytics/actions/runs/17724110223): `docker.yml` run 1132

**Comment by glenn-jocher** [comment]
@Y-T-G nice work! But @onuralpszr for actual slow tests we need to enable the failing tests too. I'll do that now.

**Comment by glenn-jocher** [comment]
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)<sub>

GitHub Actions below triggered via workflow dispatch on this PR branch `obb-nms-gpu` at 2025-09-15 10:00:25 UTC with `@ultralytics/run-ci` command:

* ✅ [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/runs/17729261406): `ci.yml` run 12007
* ✅ [Publish Docker Images](https://github.com/ultralytics/ultralytics/actions/runs/17729261490): `docker.yml` run 1134

**Comment by UltralyticsAssistant** [comment]
Absolutely thrilled to see this merged—awesome work, @Y-T-G! Huge thanks as well to @onuralpszr and @glenn-jocher for the thoughtful contributions.

“Quality is not an act, it is a habit.” — Aristotle. This PR embodies that: robust NMS handling for OBB in ONNX and re-enabled tests mean fewer edge-case crashes, stronger CI guardrails, and smoother exports for everyone. 

Details: safe masking when no detections pass thresholds, consistent `nms=nms` in ONNX export tests, and restored OBB coverage—all without API changes. This is the kind of polish that keeps our users confident and productive.

Merged: [Fix ONNX OBB NMS edge cases and re-enable tests (PR #22079)](https://github.com/ultralytics/ultralytics/pull/22079) 🚀

Thank you for leveling up reliability across the stack—your care shows.
